src/dfx-core/src/config/model/dfinity.rs

```diff
@@ -407,19 +407,27 @@ impl CanisterTypeProperties {
     }
 }
 
-#[derive(Copy, Clone, Default, Debug, PartialEq, Eq, Serialize, Deserialize, JsonSchema)]
-#[serde(rename_all = "lowercase")]
+#[derive(Clone, Default, Debug, PartialEq, Eq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
 pub enum CanisterLogVisibility {
     #[default]
     Controllers,
     Public,
+    AllowedViewers(Vec<String>),
```

Comment:
```suggestion
    #[schemars(with = "AllowedViewers(Vec::<String>)")]
    AllowedViewers(Vec<Principal>),
```
this could save us a lot of `unwrap()`s in other places

Reply:
Fixed at https://github.com/dfinity/sdk/pull/3921/commits/07b2ce497e7de94506012f7e669ff78d5779543d.